src/lib/environment/environment_record_trait.rs

```diff
@@ -60,6 +60,11 @@ pub trait EnvironmentRecordTrait: Debug + Trace + Finalize {
     /// Return true if it does and false if it does not.
     fn has_this_binding(&self) -> bool;
 
+    fn get_this_binding(&self) -> Value {
+        debug_assert!(self.has_this_binding());
```

Comment:
not sure if the assert is needed?
It won't even compile if a `has_this_binding()` isn't implemented

Reply:
`has_this_binding` can return false for some environment records, although the interpreter should never call `get_this_binding` for them. I am asserting that invariant here.